### PR TITLE
Update recipe to build the package for both names

### LINKED ISSUE #12: Duplicate feedstock
This feedstock is essentially a duplicate repo for https://github.com/conda-forge/trio_asyncio-feedstock. Looking at the creation dates:

* conda-forge/trio-asyncio-feedstock: `"created_at": "2019-06-18T17:37:10Z"`
* conda-forge/trio_asyncio-feedstock: `"created_at": "2018-04-27T16:40:01Z"`

The underscore variant got first 😬 That said, there are more `trio-*` packages so maybe for consistency we should keep the dash version? @conda-forge/trio-asyncio @conda-forge/trio_asyncio, please reach an agreement and decide which feedstock will absorb the other one by creating the needed alias output.

Thanks!

### PR DESCRIPTION
**Pull request by dhirschfeld**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

----

Resolves #12

### CODE REVIEW

**Comment by dhirschfeld** [comment]
@conda-forge-admin, please rerender

**Comment by dhirschfeld** [comment]
@scopatz / @jaimergp - This should be good to merge!

**Comment by jaimergp** [comment]
Thanks @dhirschfeld, what's the situation with the other feedstock? Is it archived? We might need to also amend `feedstock-outputs` to allow this output here.

File: recipe/meta.yaml
Comment (jaimergp) [inline recipe/meta.yaml]: Not to bikeshed, but if it's just a matter of picking one, I think it makes more sense to keep the main output as the dashed version to match the feedstock name, and then add the other as an alias.

Otherwise, we need the `extra.feedstock-name` correction.

Comment (dhirschfeld) [reply recipe/meta.yaml]: Fair enough. I mainly did it that way to use the name template in the url but I can just hard-code it there

**Comment by dhirschfeld** [comment]
> Thanks @dhirschfeld, what's the situation with the other feedstock? Is it archived?

The other feedstock isn't archived yet - it'll need to be after this is merged.

**Comment by dhirschfeld** [comment]
@smurfix the [`trio_asyncio-feedstock`](https://github.com/conda-forge/trio_asyncio-feedstock/) is being consolidated into this feedstock. As a maintainer of the old feedstock, would you like to be added as a maintainer here?

**Comment by dhirschfeld** [comment]
> *We might need to also amend `feedstock-outputs` to allow this output here.*

Sorry, not too sure what you mean by this?

**Comment by jaimergp** [comment]
> > _We might need to also amend `feedstock-outputs` to allow this output here._
> 
> Sorry, not too sure what you mean by this?

There are two feedstocks producing one package name each:

- trio-asyncio-feedstock -> trio-asyncio
- trio_asyncio-feedstock -> trio_asyncio

With these changes, we'll have

- trio-asyncio-feedstock -> trio-asyncio, trio_asyncio
- trio_asyncio-feedstock -> [archived]

Since trio_asyncio was initially claimed by trio_asyncio, we need to add an exception to the `feedstock-outputs` repo to allow a 2nd feedstock to produce that output. If we didn't have those protections, I could start shipping malicious numpy packages from this feedstock and no one would ever know :D 

Essentially, add `trio-asyncio` to this file: https://github.com/conda-forge/feedstock-outputs/blob/main/outputs/t/r/i/trio_asyncio.json

**Comment by dhirschfeld** [comment]
TIL the `feedstock-outputs` repository! I put in a PR there also.

**Review by jaimergp — APPROVED** [review]
LGTM. The other maintainers can join the team via an automated PR whenever they need :)

**Comment by dhirschfeld** [comment]
> The other maintainers can join the team via an automated PR whenever they need :)

Yep - I'm happy to add them whenever they'd like. I think this is good to merge. It might need someone from `conda-forge/core` though as I'm not sure if Anthony is about.

**Comment by dhirschfeld** [comment]
ping @conda-forge/core!

This PR is good to merge and, with it, https://github.com/conda-forge/trio_asyncio-feedstock/blob/main/recipe/meta.yaml can be archived (do I need to add a comment to the README there or is that part of the archival process?)

**Comment by beckermr** [comment]
Please submit the archive request to admin-requests and link back to this PR.

**Comment by smurfix** [comment]
I'm sorry, but I have too much on my plate to also concern myself with maintaining Conda feedstocks. Somebody else will have to take that on.

**Comment by beckermr** [comment]
No worries!

**Comment by dhirschfeld** [comment]
> I'm sorry, but I have too much on my plate to also concern myself with maintaining Conda feedstocks. Somebody else will have to take that on.

Not a problem - `conda-forge` is a community based effort and asking if you'd like to help maintain is more to prevent you from feeling like you're being excluded. If you'd ever like to help maintain this feedstock you only need to [open an issue](https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-add-user-username) on this feedstock and I'll be sure to merge it!

**Pull request merged**
